### PR TITLE
Modal component

### PR DESCRIPTION
**Pull request by EduardTrutsyk**
### What does this PR do:
Adds the Modal component to Travix UI Kit.
screenshot at 18-04-36

### Where should the reviewer start:
* Diffs
* Look at the styleguide

### Unit and/or functional tests:
Adds Unit tests for Modal component, reaching 100% coverage

### CODE REVIEW

**Comment by mAiNiNfEcTiOn** [comment]
It wasn't that hard, was it? :) 👍

File: components/modal/modal.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+    <div>
+      <div>
+        <Button size="s" onClick={() => setState({ isOpenBaseModal: true })}>Open base modal</Button>
+        <Modal
+          active={state.isOpenBaseModal}
+          onClose={() => setState({ isOpenBaseModal: false })}>
+            Modal Content
+        </Modal>
+      </div>
```
Comment (AlexDudar) [inline components/modal/modal.md]: can you add <br/> after each modal so it's going to be more user friendly

File: components/modal/modal.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      <div>
+        <Button size="s" onClick={() => setState({ isOpenModalWithTitle: true })}>Open modal with title</Button>
+        <Modal
+          active={state.isOpenModalWithTitle}
+          title="Modal with title"
+          closable={false}
+          closeOnOverlayClick={false}
+          closeOnEsc={false}
+          onClose={() => setState({ isOpenModalWithTitle: false })}>
+            You can't close this modal on ESC or overlay click
```
Comment (AlexDudar) [inline components/modal/modal.md]: ESC doesn't work for me

File: components/modal/modal.md (hunk 1 above)
Comment (mAiNiNfEcTiOn) [inline components/modal/modal.md]: Can you add what?

Comment (AlexDudar) [inline components/modal/modal.md]: `<br>`

File: components/global/global.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ * Global component
+ * React component for transportation of modals, lightboxes, loading bars... to document.body
+ */
+class Global extends Component {
+  componentDidMount() {
+    if (this.props.noscroll) {
+      global.window.document.body.classList.add('ui-global_noscroll');
+    }
+
+    this.modalTarget = global.window.document.createElement('div');
```
Comment (asci) [inline components/global/global.js]: why it is named `modalTarget` if this class supposed to serve not only modals?

File: components/modal/modal.js [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+      setTimeout(() => this.setState({ isOpen: false }), 300);
+    });
+  }
+
+  handleClose = (e) => {
+    this.close(e);
+  };
+
+  handleKeydown = (e) => {
+    if (e.keyCode === 27) {
```
Comment (asci) [inline components/modal/modal.js]: would be nice to have constant for that

File: themes/_default.yaml [hunk 5]
```diff
@@ -1,4 +1,6 @@
 generic:
+  breakpoints:
+    small-max: '768px'
```
Comment (iwwwi) [inline themes/_default.yaml]: Breakpoints should not be part of the ui kit. Can you pls provide your solution without using it? :)

File: components/modal/modal.js [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+    if (closeOnOverlayClick) {
+      this.close(e);
+    }
+  };
+
+  renderHeader() {
+    const { title } = this.props;
+    if (!title) {
+      return null;
+    }
```
Comment (iwwwi) [inline components/modal/modal.js]: I would separate the `if` conditions with blank lines to have clear indication that is a separate block :)

File: components/global/global.js (hunk 3 above)
Comment (EduardTrutsyk) [inline components/global/global.js]: yep, firstly I implemented it for modal, and then refactor, will rename.

File: components/modal/modal.js (hunk 4 above)
Comment (EduardTrutsyk) [inline components/modal/modal.js]: yep, will create common constants for that

File: themes/_default.yaml (hunk 5 above)
Comment (EduardTrutsyk) [inline themes/_default.yaml]: hmm, what? it's a variable, and it's easy to override, no?

**Pull request merged**